transformer/dockerfilegenerator/nodejsdockerfiletransformer.go

```diff
@@ -118,6 +163,54 @@ func (t *NodejsDockerfileGenerator) DirectoryDetect(dir string) (services map[st
 	return services, nil
 }
 
+// getNodeVersion returns the Node version to be used for the service
+func getNodeVersion(node string, allowedNodeVersions []int) string {
```

Comment:
Use https://github.com/hashicorp/go-version instead

Reply:
The problem with using a library for parsing in this case is that, suppose if the version in package.json is `<16.15`, then I think that the library-based parser would return the version as 15. But, `registry.redhat.io/rhel8/nodejs-15` image does not exist in the Red Hat certified container images and the versions which are available are 10, 12, 14 and 16.

Reply:
Example: https://go.dev/play/p/vdKWgIP0IwB